### PR TITLE
DPE-2655 profile limit option

### PR DESCRIPTION
**Pull request by paulomach**
port of k8s [pr#283](https://github.com/canonical/postgresql-k8s-operator/pull/283)

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -1361,10 +1367,13 @@ def _is_workload_running(self) -> bool:
     def update_config(self, is_creating_backup: bool = False) -> bool:
         """Updates Patroni config file based on the existence of the TLS files."""
         enable_tls = all(self.tls.get_tls_files())
+        limit_memory = None
+        if self.config.profile_limit_memory:
+            limit_memory = self.config.profile_limit_memory * 10**6
```
Comment (marceloneppel) [inline src/charm.py:1372]: Should we use `1024 * 1024` instead of `10**6` as the config description states `MegaBytes`?

Comment (paulomach) [reply src/charm.py:1372]: I think MB to B is `10**6` or am I wrong?

**Review by marceloneppel — APPROVED** [review]

File: src/charm.py (hunk 1 above)
Comment (dragomirp) [reply src/charm.py:1372]: Does the docs specify if it's MB or MiB?

**Review by dragomirp — APPROVED** [review]

File: src/charm.py (hunk 1 above)
Comment (paulomach) [reply src/charm.py:1372]: Spec is MB, and pg config knows how to handle different suffixes. Charm calculations being done in bytes

**Pull request merged**